lib/glific/flows/broadcast.ex

```diff
@@ -329,18 +336,60 @@ defmodule Glific.Flows.Broadcast do
     Map.put_new(map, :failed, count)
   end
 
-  defp count_failed_deliveries_by_category(map, flow_broadcast_id) do
-    Map.put(map, :failed_catogries, failed_deliveries_by_category(flow_broadcast_id))
+  @spec count_deliveries_by_category(map(), non_neg_integer()) :: map()
+  defp count_deliveries_by_category(map, flow_broadcast_id) do
+    Map.put(map, :msg_categories, msg_deliveries_by_category(flow_broadcast_id))
   end
 
-  defp failed_deliveries_by_category(flow_broadcast_id) do
+  @spec msg_deliveries_by_category(non_neg_integer()) :: map()
+  defp msg_deliveries_by_category(flow_broadcast_id) do
     data =
       broadcast_stats_base_query(flow_broadcast_id)
       |> Repo.query!()
 
+    sent_count =
+      Enum.count(data.rows, fn d ->
+        [_message_id, message_status, _processed_at, _broadcast_status, _bsp_status, _errors] = d
+        message_status == "sent"
+      end)
+
+    read_count =
+      Enum.count(data.rows, fn d ->
+        [_message_id, _message_status, _processed_at, _broadcast_status, bsp_status, _errors] = d
+        bsp_status == "read"
+      end)
+
+    delivered_count =
+      Enum.count(data.rows, fn d ->
+        [_message_id, _message_status, _processed_at, _broadcast_status, bsp_status, _errors] = d
+        bsp_status == "delivered"
+      end)
+
+    opted_out_count =
```

Comment:
I did not get this one. I don't think we are processing the opt-out and blocked contacts. 
Check the code in broadcast.ex file.

Reply:
@pankaj-ag I overlooked this as it was already there before I made changes to the file. Also, there are a couple of messages in DB where the status in the message table is `contact_opt_out`
I have removed it now after testing sending message from broadcast to a contact who is opted out and we don't create message for it